### PR TITLE
Faillock: Use pluralization via dngettext or fallback

### PR DESCRIPTION
**Pull request by Razzeee**
Haven't done C in a long time, so I might need some help / CI checks with this :)

Fixes https://github.com/linux-pam/linux-pam/issues/322#issuecomment-764192605

### CODE REVIEW

File: modules/pam_faillock/pam_faillock.c [hunk 1]
```diff
@@ -620,7 +620,17 @@ faillock_message(pam_handle_t *pamh, struct options *opts)
...
 			left = (left + 59)/60; /* minutes */
 
-			pam_info(pamh, _("(%d minutes left to unlock)"), (int)left);
+#if defined HAVE_DNGETTEXT && defined ENABLE_NLS
+		pam_info(pamh, _(dngettext(PACKAGE,
+		"(%d minute left to unlock)",
+		"(%d minutes left to unlock)",
+		(int)left)));
+#else
+	if ((int)left == 1)
```
Comment (t8m) [inline modules/pam_faillock/pam_faillock.c:629]: No need to use the (int) cast here.

File: modules/pam_faillock/pam_faillock.c [hunk 2]
```diff
@@ -620,7 +620,17 @@ faillock_message(pam_handle_t *pamh, struct options *opts)
 		if (left > 0) {
 			left = (left + 59)/60; /* minutes */
 
-			pam_info(pamh, _("(%d minutes left to unlock)"), (int)left);
+#if defined HAVE_DNGETTEXT && defined ENABLE_NLS
+		pam_info(pamh, _(dngettext(PACKAGE,
```
Comment (t8m) [inline modules/pam_faillock/pam_faillock.c:624]: Please drop the `_(` here.

File: modules/pam_faillock/pam_faillock.c [hunk 3]
```diff
@@ -620,7 +620,17 @@ faillock_message(pam_handle_t *pamh, struct options *opts)
...
+		"(%d minute left to unlock)",
+		"(%d minutes left to unlock)",
+		(int)left)));
```
Comment (t8m) [inline modules/pam_faillock/pam_faillock.c:627]: This line needs to look like this: `(int)left), (int)left);`

**Review by t8m — CHANGES_REQUESTED** [review]
Needs some changes. Also the indentation looks wrong.

File: modules/pam_faillock/pam_faillock.c (hunk 1 above)
Comment (Razzeee) [reply modules/pam_faillock/pam_faillock.c]: Changed

File: modules/pam_faillock/pam_faillock.c (hunk 2 above)
Comment (Razzeee) [reply modules/pam_faillock/pam_faillock.c]: Dropped

File: modules/pam_faillock/pam_faillock.c (hunk 3 above)
Comment (Razzeee) [reply modules/pam_faillock/pam_faillock.c]: Thanks, already figured that out due to the CI :)

File: modules/pam_faillock/pam_faillock.c [hunk 4]
```diff
@@ -620,7 +620,20 @@ faillock_message(pam_handle_t *pamh, struct options *opts)
...
+		dngettext(PACKAGE,
+			"(%d minute left to unlock)",
+			"(%d minutes left to unlock)",
+			(int)left),
+		(int)left);
+#else
+	if (left == 1)
+		pam_info(pamh, _("(%d minute left to unlock)"), (int)left);
+	else
+		pam_info(pamh, _("(%d minutes left to unlock)"), (int)left);
```
Comment (t8m) [inline modules/pam_faillock/pam_faillock.c:635]: You still need to add 2 tabs of indentation everywhere.

Comment (Razzeee) [reply modules/pam_faillock/pam_faillock.c]: Good catch, done.

**Comment by t8m** [comment]
I think this can be now pulled out of the Draft status.

**Comment by Razzeee** [comment]
Done

**Review by t8m — APPROVED** [review]

File: modules/pam_faillock/pam_faillock.c [hunk 5]
```diff
@@ -620,7 +620,20 @@ faillock_message(pam_handle_t *pamh, struct options *opts)
...
+				dngettext(PACKAGE,
+					"(%d minute left to unlock)",
+					"(%d minutes left to unlock)",
+					(int)left),
+				(int)left);
+#else
+			if (left == 1)
+				pam_info(pamh, _("(%d minute left to unlock)"), (int)left);
+			else
+				pam_info(pamh, _("(%d minutes left to unlock)"), (int)left);
```
Comment (ldv-alt) [inline modules/pam_faillock/pam_faillock.c:635]: Please add a comment for translators:
/* TRANSLATORS: only used if dngettext is not supported.  */

**Review by ldv-alt — CHANGES_REQUESTED** [review]
Commit message:
- please lowercase the first line
- please fix typo: "dngtext" -> "dngettext".

File: modules/pam_faillock/pam_faillock.c (hunk 5 above)
Comment (Razzeee) [reply modules/pam_faillock/pam_faillock.c:635]: Probably also for line 633?

**Review by t8m — APPROVED** [review]

**Review by ldv-alt — APPROVED** [review]

**Comment by t8m** [comment]
Merged. Thank you for your contribution.

**Pull request merged**